### PR TITLE
Rails5

### PR DESCRIPTION
**Pull request by PamMarie**
Cleanup and rails5 compatibility changes. 

Update before_filter calls to before_action
Remove railtie
Use to_unsafe_h to convert ActionController:Parameter to Hash for ease-of-use

### CODE REVIEW

**Comment by hiramkumar** [comment]
Hi, When this PR will merge? We are upgrading our existing Project to Rails 5.2, Please merge as soon as possible.

**Comment by PamMarie** [comment]
@hiramkumar You should be able to use this release tag for your system: https://github.com/InfoTech/salesforce_ar_sync/releases/tag/v3.0.0

File: salesforce_ar_sync.gemspec [hunk 1]
```diff
@@ -1,29 +1,28 @@
...
-require File.expand_path('../lib/salesforce_ar_sync/version', __FILE__)
+require File.expand_path('lib/salesforce_ar_sync/version', __dir__)
 
 Gem::Specification.new do |gem|
-  gem.authors       = ["Michael Halliday", "Nick Neufeld", "Andrew Coates", "Devon Noonan", "Liam Nediger"]
-  gem.email         = ["mhalliday@infotech.com", "nneufeld@infotech.com", "acoates@infotech.com", "dnoonan@infotech.com", "lnediger@infotech.com"]
-  gem.description   = %q{ActiveRecord extension & rails engine for syncing data with Salesforce.com}
-  gem.summary       = %q{ActiveRecord extension & rails engine for syncing data with Salesforce.com}
-  gem.homepage      = "http://github.com/InfoTech/"
+  gem.authors       = ['Michael Halliday', 'Nick Neufeld', 'Andrew Coates', 'Devon Noonan', 'Liam Nediger']
```
Comment (jbrodie) [inline salesforce_ar_sync.gemspec]:
```suggestion
  gem.authors       = ['Liam Nediger']
```

File: salesforce_ar_sync.gemspec [hunk 2]
```diff
@@ -1,29 +1,28 @@
...
+  gem.email         = ['mhalliday@infotech.com', 'nneufeld@infotech.com', 'acoates@infotech.com', 'dnoonan@infotech.com', 'lnediger@infotech.com']
```
Comment (jbrodie) [inline salesforce_ar_sync.gemspec]:
```suggestion
  gem.email         = ['lnediger@infotech.com']
```

File: lib/salesforce_ar_sync/salesforce_sync.rb [hunk 3]
```diff
@@ -118,101 +120,96 @@ def salesforce_attributes_to_set(attributes = {})
...
-      self.save!
+      save!
     end
 
-#    def salesforce_object_exists?
-#      return salesforce_object_exists_method if respond_to? salesforce_exists_method
-#      return salesforce_object_exists_default
-#    end
-
+    #    def salesforce_object_exists?
```
Comment (jbrodie) [inline lib/salesforce_ar_sync/salesforce_sync.rb]: 💣

**Comment by hiramkumar** [comment]
> https://github.com/InfoTech/salesforce_ar_sync/releases/tag/v3.0.0

Thanks @PamMarie

**Pull request closed without merging**